upgrade/UPGRADE-v9.0.0-dev.md

```diff
@@ -348,82 +285,126 @@ There you can find links to upgrade notes for other versions too.
-               {{ getShopInfoPhoneNumber() }}
-               {{ getShopInfoEmail() }}
-               {{ render(controller('App\\Controller\\Front\\ArticleController:footerAction')) }}
-        +      <a class="menu__item__link" href="{{ url('front_contact') }}">{{ 'Contact'|trans }}</a>
-            </div>
+        -   {% if product.isMainVariant %}
+        +   {% if product.isMainVariant and not product.calculatedSellingDenied %}
         ```
 
 - vats can be created and managed per domains ([#1498](https://github.com/shopsys/shopsys/pull/1498))
     - please read [upgrade instruction for vats per domain](https://github.com/shopsys/shopsys/blob/master/upgrade/upgrade-instruction-for-vats-per-domain.md)
 
 - apply these changes to add support for naming uploaded files([#1547](https://github.com/shopsys/shopsys/pull/1547))
-    - update your `composer.json`
-        ```diff
-            "ext-curl": "*",
-        +   "ext-fileinfo": "*",
-            "ext-gd": "*",
-        ```
-    - update your `docker/php-fpm/Dockerfile`
-        ```diff
-            RUN docker-php-ext-install \
-                bcmath \
-        +       fileinfo \
-                gd \
-        ```
-    - add this route to the end of your `config/routes/shopsys_front.yaml`
-        ```diff
-        +   front_download_uploaded_file:
-        +       path: /file/{uploadedFileId}/{uploadedFilename}
-        +       defaults: { _controller: App\Controller\Front\UploadedFileController:downloadAction }
-        +       methods: [GET]
-        +       requirements:
-        +           uploadedFileId: \d+
-        ```
-    - update your `src/Controller/Front/OrderController.php`
-        ```diff
-             return new DownloadFileResponse(
-                 $this->legalConditionsFacade->getTermsAndConditionsDownloadFilename(),
-        -        $response->getContent()
-        +        $response->getContent(),
-        +        'text/html'
-             );
-        ```
-    - update your `tests/App/Smoke/Http/RouteConfigCustomization.php`
-        ```diff
-                $config->addExtraRequestDataSet('Check personal data XML export with right hash')
-                    ->setParameter('hash', $personalDataAccessRequest->getHash())
-                    ->setExpectedStatusCode(200);
-        +   })->customizeByRouteName(['front_download_uploaded_file'], function (RouteConfig $config) {
-        +       $config->skipRoute('Downloading uploaded files is not tested.');
-            });
-        ```
-    - add [src/Controller/Front/UploadedFileController.php](https://github.com/shopsys/shopsys/blob/master/project-base/src/Controller/Front/UploadedFileController.php) to your project
+    - see [project-base diff](https://github.com/shopsys/project-base/commit/1a2fb6dff9f111ed36c91114d00d45fe9054baaa) to update your project
+
     - `MessageData::attachmentsFilepaths` has been replaced by `MessageData::attachments` that accepts array of `UploadedFile`
```

Comment:
Is is it possible to replace below instructions about UploadedFile within project-based diff ?

Reply:
The project-base diff is present, there is the only a list of changed method interfaces in the framework. Users should check if he is not using any of these methods in the project and if so, fix theirs calls.